pkgs/development/ocaml-modules/plotkicadsch/open-process.patch

```diff
@@ -0,0 +1,19 @@
+diff --git a/plotkicadsch/src/sysAbst.ml b/plotkicadsch/src/sysAbst.ml
+index c3c7b52..4300e8f 100644
+--- a/plotkicadsch/src/sysAbst.ml
++++ b/plotkicadsch/src/sysAbst.ml
+@@ -30,9 +30,9 @@ let detect_os () : os =
+   if Sys.win32 then Windows
+   else if Sys.cygwin then Cygwin
+   else
+-    let ((in_ch, _, _) as uname) = UnixLabels.open_process_full "uname" ~env:[| |] in
```

Comment:
this code fails on nixos because `uname` is only looked in /bin and /usr/bin when PATH is empty. It would be better to patch the code to use ${coreutils}/bin/uname instead.

Reply:
What about a wrapper?

Reply:
@leungbk `uname` is part of coreutils. Wrapper [example](https://github.com/NixOS/nixpkgs/blob/350fd0044447ae8712392c6b212a18bdf2433e71/pkgs/tools/package-management/nix-bundle/default.nix#L39).

Reply:
a wrapper will not work as they empty the env

Reply:
> It would be better to patch the code to use ${coreutils}/bin/uname instead.

Done.